arch/x86/link.ld

```diff
@@ -89,6 +89,13 @@ SECTIONS
         __end_data_init = .;
     } :init
 
+    .cmdline.init BLOCK(4K) : ALIGN(4K)
+    {
+        __start_cmdline_init = .;
+            *(.cmdline.init)
+        __end_cmdline_init = .;
+    } :init
+
```

Comment:
I would not call it *.init and would rather place it in the `:kernel` segment

I would actually prefer to put it into `.data`:

```    .data BLOCK(4K) : ALIGN(4K)
    {
        . = ALIGN(4K);
        __start_data = .;
            *(.data)
        __start_cmdline = .;
            *(.cmdline)
        __end_cmdline = .;
        . = ALIGN(4K);
        __end_data = .;
    } :kernel
```